packages/dataviews/src/types.ts

```diff
@@ -536,12 +536,21 @@ export type NormalizedCombinedFormField< Item > = CombinedFormField< Item > & {
 	Edit?: ComponentType< DataFormCombinedEditProps< Item > >;
 };
 
+export type FormField =
```

Comment:
Based on what I saw so far, this is the API I was expecting for `FormField`:

```ts
export type FormField =
  | string // The field id
  | {
      id: string; // The field id
      layout?: '...';
      fields?: FormField[]; // Allows grouping fields
  };
```

Some thoughts:

- Why do we need `field` in `FormField`?
- Can the form.combinedFields API be removed? Why do we need both?

Reply:
I reckon some of this is pre-existing to this PR. If you think it's too much for this PR, we could do a PR to simplify the implementation first, and then rebase this PR to use that.

Reply:
> Why do we need field in FormField?

Actually my thought was to use `field` for the `field` id instead of `id`, as `field` is more clear. We did probably want to replace `fields` with `children` to not make it to confusing.
Mostly for this scenario, where what would you do if you wanted to display a horizontal column layout within a panel. Or any horizontal layout in general. In this case we are just grouping fields, without needing to reference a specific field.
It may look like this:
```
{
  id: "status",
  layout: "panel",
  fields: [
      {
          layout: 'horizontal' // if we go this route
          fields: [ 'status', 'password' ]
      }
  ]
}
```
We may still want to add an `id` here as a `key`, although we could also generate one.

> Can the form.combinedFields API be removed? Why do we need both FormField and combinedFields?

Yeah `combinedFields` can be removed, as this API replaces that functionality. I was thinking of doing that as a follow up. But happy to remove it as part of this PR.

Reply:
> We did probably want to replace fields with children to not make it to confusing.

Agree to this, it'd make the code clearer.

Why the `regular` layout can't take children but the `panel` layout can? The existing combinedFields API worked for both.

Reply:
Another thought is: why only the fields that are defined via object in `form.fields` should have the `layout` property? It looks like this is optimizing for the current use case (combining status + password), but I wonder if the layout should be part of the Fields API instead?